api_app/analyzers_manager/file_analyzers/cape_sandbox.py

```diff
@@ -0,0 +1,145 @@
+import time
+
+import requests
+
+from api_app.analyzers_manager.classes import AnalyzerRunException, FileAnalyzer
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+
+class CAPEsandbox(FileAnalyzer):
+    max_tries: int = 25
+    to_respond = {}
+
+    def set_params(self, params):
+        self.__token = self._secrets["token"]
+        self.__vm_name = params.get("VM_NAME", "win7x64_8")
+        self.BASE_URL = self._secrets.get("url_key_name", "https://www.capesandbox.com")
+
+    def run(self):
+        API_URL: str = self.BASE_URL + "/apiv2/tasks/create/file/"
+        self.headers = {
+            "Authorization": f"Token {self.__token}",
+        }
+
+        files = {
+            "file": (self.filename, self.read_file_bytes()),
+        }
+
+        data = {"machine": self.__vm_name}
+
+        try:
+            response = requests.post(
+                API_URL, headers=self.headers, files=files, data=data
+            )
+            response_json = response.json()
+            response.raise_for_status()
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+        # since not None == True as well. This is a more precise way.
+        if response_json.get("error") is False:
+            self.to_respond["result_url"] = response_json.get("url")
+            self.task_id = response_json.get("data").get("task_ids")[0]
+            self.status_api = self.BASE_URL + "/apiv2/tasks/status/" + str(self.task_id)
```

Comment:
pass the `task_id` as an argument to `__poll_for_result` and build the `status_api` URL inside that function.

We should avoid setting a variable to `self` as much as possible.